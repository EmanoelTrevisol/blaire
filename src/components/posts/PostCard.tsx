import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
// import { Container } from './styles';

const Detail = (props) => {
  const { title, body, username, createdAt } = props;

  return (
    <View style={stl.card}>
      <View style={stl.header}>
        <View style={stl.headerTitle}>
          <Text style={stl.headerTitleText}>{title}</Text>
          <Text style={stl.headetTitleSubText}>{createdAt}</Text>
        </View>
        <View style={stl.headarSub}>
          <Text style={stl.headerSubText}>{username}</Text>
        </View>
      </View>
      <View style={stl.body}>
        <Text>{body}</Text>
      </View>
    </View>
  );
};

const stl = StyleSheet.create({
  card: {
    backgroundColor: '#fff',
    paddingVertical: 10,
    marginHorizontal: 15,
    marginBottom: 15,
    borderRadius: 20,
    elevation: 10,
    shadowColor: '#000',
    shadowRadius: 3,
    shadowOpacity: 0.2,
    shadowOffset: {
      width: 0,
      height: 4,
    },
  },
  header: {
    padding: 15,
    borderBottomColor: '#DADADA',
    borderBottomWidth: 1,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  headerTitle: {
    maxWidth: '60%',
  },
  headerTitleText: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 3,
  },
  headetTitleSubText: {
    fontSize: 12,
    color: '#444',
  },
  headarSub: {
    maxWidth: '35%',
    marginLeft: 10,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  headerSubText: {
    textAlign: 'center',
    flex: 1,
    flexWrap: 'wrap',
  },
  body: {
    padding: 15,
  },
});

export default Detail;
