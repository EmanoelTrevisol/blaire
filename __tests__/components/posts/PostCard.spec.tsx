import React from 'react';
import PostCard from '@/components/posts/PostCard';
import { render } from '@testing-library/react-native';

let props;

describe('PostCard.tsx', () => {
  const renderComp = (cpProps) => {
    return render(<PostCard {...cpProps} />);
  };

  beforeEach(() => {
    props = {
      title: 'Título de exemplo',
      body: 'Esse é o post. É pequeno pq é só um exemplo',
      id: 'asdasdadadadsadasdasd',
      createdAt: 1605399247782,
      updatedAt: 1605399247782,
    };
  });

  test('matches snapshot', () => {
    const { toJSON } = renderComp(props);

    expect(toJSON()).toMatchSnapshot();
  });
});
