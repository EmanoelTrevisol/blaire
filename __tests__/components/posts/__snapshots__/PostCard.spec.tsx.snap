// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard.tsx matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 20,
      "elevation": 10,
      "marginBottom": 15,
      "marginHorizontal": 15,
      "paddingVertical": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#DADADA",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "maxWidth": "60%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 3,
          }
        }
      >
        Título de exemplo
      </Text>
      <Text
        style={
          Object {
            "color": "#444",
            "fontSize": 12,
          }
        }
      >
        1605399247782
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 10,
          "maxWidth": "35%",
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
            "textAlign": "center",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 15,
      }
    }
  >
    <Text>
      Esse é o post. É pequeno pq é só um exemplo
    </Text>
  </View>
</View>
`;
